(feat) Improve EditTeam flow

brrkrmn opened the pull request:
## Summary

The EditTeam view's code involved unused functions and variables, and the design was not consistent with the EditProfile view. Additionally, I implemented the code to show error message when the user tries to submit the form without a  team name. 

## Changes for the appearance

- Edit Team button label is changed from "Edit Team Info" to "Edit Team"
- The input headings were removed since the labels convey the same text
- A description text is added for the bio field
- Validation is added so that empty username field shows an error
- Changed the submit button's label from "Edit Profile" to "Save Changes"
- Add a button to discard changes
- Edited delete team button style and removed the description since it is shown inside the modal

PS: I used hardcoded texts for some labels and Typography elements. This is because in my other PR #843, I handled all the hardcoded texts to be used from the translation files and created new keys for this file. I could not use those keys because the PR may not be merged. I also did not want to do the same steps twice not waste time. I will change the texts of EditTeam.jsx and EditTeamScripts.js as soon as that #843 is resolved.

## Changes regarding code refactoring

I noticed that the code in EditTeam.jsx and EditTeamScripts.js is copied from the EditProfile view and has many parts that should be changed or removed. These are the changes to simplify and refactor this code:

- EditProfile form's refs were deleted and two new refs for EditTeam form is created
- Unused functions and variables were removed
- In the old code, form values were saved in the state manually. Now, Formik's `handleChange` prop is used. Thus, the old state is removed.
- The handleSubmit function is moved to EditTeamScripts.js EditProfile function so that the form submits after all the validations run.
- New id's and names were given to the form elements because the existing id's and refs caused the bio field to get the data from the user's profile. 
- To prevent previous complications to happen, `getProfile()` function in the effect is removed because the code does not use any of its return values.
- Edited getTeamProfile function to fill the username and bio fields if they have an initial value
- Validations are added for the form elements

## The backend change

After using the Formik props, the form did not submit when the user left the bio field empty. In the frontend, the bio field is sent to the backend as empty string if the user sent it empty. The form worked as expected whwn the bio is filled. When I checked edit profile form's bio field, it is also sent as an empty string when submitted empty. Thus, I could not prevent this from happening with changes to the frontend. 

https://stackoverflow.com/a/55150513 => Assuming that the problem is caused because Django interprets empty string as falsy, I changed the line to raise a validation error only if the groupname is missing.

I am not really sure about this solution, looking forward to reviews 🙂

## Videos

Before

https://github.com/unstructuredstudio/zubhub/assets/92817363/2405c0b7-c382-4a49-b722-989052be7c4a


After

https://github.com/unstructuredstudio/zubhub/assets/92817363/209e3693-3a21-4207-af22-69a7ab9e2daa

brrkrmn committed:
initial

brrkrmn committed:
Merge branch 'master' of https://github.com/brrkrmn/zubhub

brrkrmn committed:
Merge branch 'master' of https://github.com/brrkrmn/zubhub

brrkrmn committed:
Merge branch 'master' of https://github.com/brrkrmn/zubhub

brrkrmn committed:
Merge branch 'master' of https://github.com/brrkrmn/zubhub

brrkrmn committed:
Merge branch 'unstructuredstudio:master' into master

brrkrmn committed:
revert package-lock.json

brrkrmn committed:
fix package-lock.json

brrkrmn committed:
fix package-lock.json

brrkrmn committed:
show error message if team name is empty

brrkrmn committed:
change edit team form's appearance

brrkrmn committed:
change edit team form to use Formik props and unique id's

brrkrmn committed:
add buttons for canceling changes and deleting team

brrkrmn committed:
change edit team button label

brrkrmn committed:
small fixes

brrkrmn commented:
For future improvement, some functions' names can be changed to be more appropriate for the view. Instead of editProfile => editTeam etc.

brrkrmn committed:
Merge branch 'unstructuredstudio:master' into master

coderatomy commented on zubhub_frontend/zubhub/src/views/edit_team/EditTeam.jsx:
```diff
@@ -308,47 +257,63 @@ function EditTeam(props) {
+                        customButtonStyle
+                        fullWidth
+                      >
+                        {'Discard Changes'}
```
Consider translating this

brrkrmn replied:
Actually I was waiting for another PR to be merged to translate the texts. That is merged, so I am working on translating these as well, thank you! 🙂

coderatomy commented on zubhub_frontend/zubhub/src/views/edit_team/EditTeam.jsx:
```diff
@@ -308,47 +257,63 @@ function EditTeam(props) {
+                      handleSetState(handleToggleDeleteAccountModal(state))
+                    }
+                  >
+                    {'Delete Team'}
```
same here

coderatomy reviewed (changes requested):
Nice work @brrkrmn. Just a few thing you missed out
Also consider resolving the conflicts

brrkrmn committed:
merge

brrkrmn committed:
replace texts with translations

brrkrmn commented:
@shrish, I replaced the texts with translations. This is ready for review ☺️

tuxology commented:
@brrkrmn Please refer the Figma design to see edit team implementation:

![image](https://github.com/unstructuredstudio/zubhub/assets/3373857/bbe14adf-acf4-417b-a9ef-8ef05a9b0a16)
 

You can use existing styles components from what is being done in #778 I agree to your other points about code refactoring, but lets take correct components from similar styled pages such as Create Team or from work done in #778. In addition, ensure this works in responsive mode as shown in screenshots

brrkrmn commented:
@tuxology Hey, I am currently working on this according to the Figma design. I see that editTeam.jsx's styles are imported from efitProfileStyles.js. Should I create a new editTeamStyles.js file to separate the view's design?

brrkrmn committed:
change style

brrkrmn commented:
@tuxology, I've changed the files according to the Figma design. 
I created an editTeamStyles.js file, which is the duplicate of the existing styles except for the unused parts. I also added new classes for the dialog part. 
In the Figma design, there is not a submit button but the error handling and validation in this form requires one, so I kept the submit button. 
Here is the final version:
<img width="562" alt="Screenshot 2023-10-16 at 17 48 04" src="https://github.com/unstructuredstudio/zubhub/assets/92817363/ed5fb99b-bbd7-43dc-bb80-4d41610b129b">
<img width="714" alt="Screenshot 2023-10-16 at 17 48 25" src="https://github.com/unstructuredstudio/zubhub/assets/92817363/edce31a8-7215-4b35-b363-03c32708c980">

And the mobile version:
<img width="387" alt="Screenshot 2023-10-16 at 17 47 33" src="https://github.com/unstructuredstudio/zubhub/assets/92817363/00912d14-a16e-4b11-9d5b-b7f88937fe1f">
<img width="387" alt="Screenshot 2023-10-16 at 17 47 40" src="https://github.com/unstructuredstudio/zubhub/assets/92817363/ff8a915d-b871-49a9-9c9e-859ddead2dc1">

brrkrmn committed:
edit dialog actions button size

tuxology reviewed (approved):
LGTM 🌈

tuxology reviewed (changes requested):
Upon closer look, have a few comments: 

Is it possible to use https://github.com/unstructuredstudio/zubhub/blob/master/zubhub_frontend/zubhub/src/components/modals/Modal.jsx#L16 for your modal instead of creating new styles? If you would like to change the design, its fine, lets change it in a single place. I am re-marking the PR for changes.

brrkrmn committed:
use modal component for dialog

brrkrmn commented:
@tuxology I used our Modal component for the dialog. I moved the dialog styles to the component's style file but did not style the component itself to avoid changing other modals in the app. This view now uses classes from the modal component, but for reusability, I think some changes can be made to the modal component in terms of styling.
